### PR TITLE
Access to attendance sheets for parents #3093

### PR DESCRIPTION
**Pull request by carlangas159**
Added search for all student attendance to courses like RH.

They are adjusted in ascending order with a link to the course or session.

Changed functions in attendance.lib.php:

get_users_attendance_sheet
get_attendance_calendar

Added getCoursesWithAttendance

to allow SQL search and not overload for loops. By default it takes the last week

### CODE REVIEW

**Comment by ywarnier** [comment]
The syntax looks good. I didn't go into the queries themselves, but I tested it and the results seem correct.
We'll have to watch the efficiency on large portals, and make sure the link is available somewhere more practical for the DRH/parent.
Good work!

File: main/inc/lib/attendance.lib.php [hunk 1]
```diff
@@ -1216,34 +1216,55 @@ public function get_faults_average_by_course(
...
         $attendanceId,
         $user_id = 0,
-        $groupId = 0
+        $groupId = 0,
+        $course_id = 0,
+        DateTime $startDate = null,
+        DateTime $endDate = null
     ) {
+        //Get actual course or by course_id
+        $course_id = (0 == $course_id) ? api_get_course_int_id() : $course_id;
```
Comment (jmontoyaa) [inline main/inc/lib/attendance.lib.php:1239]: Falta agregar el casting para proteger los queries. :eyes: 

`(int) $course_id; `

File: main/inc/lib/attendance.lib.php [hunk 2]
```diff
@@ -1448,16 +1473,27 @@ public function get_attendance_calendar(
...
+        $course_id = 0,
+        DateTime $startDate = null,
+        DateTime $endDate = null
     ) {
         $tbl_attendance_calendar = Database::get_course_table(TABLE_ATTENDANCE_CALENDAR);
         $tbl_acrg = Database::get_course_table(TABLE_ATTENDANCE_CALENDAR_REL_GROUP);
         $attendanceId = intval($attendanceId);
-        $course_id = api_get_course_int_id();
-
+        $course_id = (0 == $course_id) ? api_get_course_int_id() : $course_id;
```
Comment (jmontoyaa) [inline main/inc/lib/attendance.lib.php:1484]: Int casting? (int) $course_id

File: main/inc/lib/attendance.lib.php [hunk 3]
```diff
@@ -2256,4 +2293,207 @@ public function exportAttendanceLogin($startDate, $endDate)
...
+
+        /* Sessions */
+        $sql = "
+            SELECT
+                session_id,
+                c_id
+            FROM
+                session_rel_course_rel_user
+            WHERE
+                user_id = $studentId";
```
Comment (jmontoyaa) [inline main/inc/lib/attendance.lib.php:2415]: :eyes: $studentId no es limpio, agregar $studentId = (int ) $studentId;' al inicio...

File: main/mySpace/myStudents.php [hunk 4]
```diff
@@ -970,6 +982,115 @@
...
+    }
+    $startDateText = api_get_local_time($startDate);
+    $endDateText = api_get_local_time($endDate);
+    // Varible for all attendance list
+
+    /** Start date and end date*/
+    $defaults['startDate'] = $startDateText;
+    $defaults['endDate'] = $endDateText;
+    $form = new FormValidator('all_attendance_list', 'GET',
+        'myStudents.php?action=all_attendance&student='.$_GET['student'].'&startDate='.$defaults['startDate'].'&endDate='.$defaults['endDate'].'&&'.api_get_cidreq(),
```
Comment (jmontoyaa) [inline main/mySpace/myStudents.php:1016]: Falta limpiar $_GET['student']

Comment (ywarnier) [reply main/mySpace/myStudents.php:1016]: Sí, y aquí si es que siempre es un entero, bastaría con un casting (int) (no es necesario en este caso usar Security::remove_XSS())

Comment (ywarnier) [reply main/mySpace/myStudents.php:1016]: Extraño, porque todo alrededor usa la versión ya filtrada $student_id.
Lo estoy corrigiendo y mandando.

**Pull request merged**